source/Ship.cpp

```diff
@@ -1586,10 +1586,14 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
 					Jettison(it.first, Random::Binomial(it.second, .25));
 				for(const auto &it : cargo.Outfits())
 					Jettison(it.first, Random::Binomial(it.second, .25));
-				// Ammunition has a 5% chance to survive as flotsam
+				// Ammunition has a default 5% chance to survive as flotsam.
 				for(const auto &it : outfits)
-					if(it.first->Category() == "Ammunition")
+				{
+					if(it.first->Get("flotsam chance") > 0.)
+						Jettison(it.first, Random::Binomial(it.second, it.first->Get("flotsam chance")));
```

Comment:
```suggestion
					double flotsamChance = it.first->Get("flotsam chance");
					if(flotsamChance)
						Jettison(it.first, Random::Binomial(it.second, flotsamChance));
```
this attribute cant be negative so no need to check for being higher than 0 being different than it is enough
also some caching